src/test/java/emailvalidator4j/parser/DomainPartTest.java

```diff
@@ -86,6 +87,8 @@ public void domainPartHasWarnings(String domainPart, List<Warnings> warnings) th
     @DataProvider
     public static Object[][] domainPartWithWarnings() {
         return new Object[][]{
+                {"@has-hyphen.example.com", Collections.emptyList()},
+                {"@1leadingnumber.example.com", Collections.emptyList()},
```

Comment:
This is a bit odd. I'm asserting that these hyphenated and number-led domains parse without warnings. I'm possibly abusing this test, but it works.

Reply:
Yes, you are :). What do you think of creating two tests with this each case for now and if more examples are needed then creating a data provider?

Reply:
I agree. I’ll split these into a separate data provider and test for valid domain parts.